src/WebJobs.Script.Grpc/Server/DefaultHttpProxyService.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Yarp.ReverseProxy.Forwarder;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    internal class DefaultHttpProxyService : IHttpProxyService
+    {
+        private IHttpForwarder _httpForwarder;
+        private HttpMessageInvoker _messageInvoker;
+        private ForwarderRequestConfig _forwarderRequestConfig;
+        private string _proxyEndpoint;
+
+        public DefaultHttpProxyService(IHttpForwarder httpForwarder)
+        {
+            _httpForwarder = httpForwarder;
+
+            var handler = new SocketsHttpHandler();
+            _messageInvoker = new HttpMessageInvoker(handler);
+            _forwarderRequestConfig = new ForwarderRequestConfig();
+
+            // TODO: Update this logic. Port should come through configuration.
+            var port = Environment.GetEnvironmentVariable("AZURE_FUNCTIONS_HTTP_PROXY_PORT") ?? "5555";
+
+            _proxyEndpoint = "http://localhost:" + port;
+        }
+
+        public ValueTask<ForwarderError> Forward(ScriptInvocationContext context)
+        {
+            HttpRequest httpRequest = context.Inputs.FirstOrDefault(i => i.Val is HttpRequest).Val as HttpRequest;
+            HttpContext httpContext = httpRequest.HttpContext;
+
+            httpContext.Items.Add("IsHttpProxying", bool.TrueString);
```

Comment:
nit: make `IsHttpProxying` a const string

Reply:
noted in [todo comment](https://github.com/Azure/azure-functions-host/blob/7efb10cc1db55dbb319b6f16e889550d4612e2d5/src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs#L67) at the other location it's used - haven't proceeded yet since this was a "hack" design decision but if we decide to keep it will do.